File: app/src/main/java/jp/gcreate/product/filteredhatebu/model/HatebuRelatedEntry.java
package jp.gcreate.product.filteredhatebu.model;

import com.google.gson.annotations.SerializedName;

/**
 * Copyright 2016 G-CREATE
 */

public class HatebuRelatedEntry {
    String title;
    int    count;
    String url;
    @SerializedName("entry_url")
    String entryUrl;
    @SerializedName("eid")
    int    entryId;

    public HatebuRelatedEntry(String title, int count, String url, String entryUrl, int entryId) {
        this.title = title;
        this.count = count;
        this.url = url;
        this.entryUrl = entryUrl;
        this.entryId = entryId;
    }

    @Override
    public String toString() {
        return "(" + getClass().getSimpleName() + ":" + hashCode() + "){"
               + "title:" + title + ","
               + "count:" + count + ","
               + "url:" + url + ","
               + "entryUrl" + entryUrl + ","
               + "id:" + entryId + "}";
    }
}

File: app/src/main/java/jp/gcreate/product/filteredhatebu/model/HatebuRelatedEntry.kt
package jp.gcreate.product.filteredhatebu.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Copyright 2016 G-CREATE
 */
@JsonClass(generateAdapter = true)
data class HatebuRelatedEntry(
    var title: String,
    var count: Int,
    var url: String?,
    @field:Json(name = "entry_url")
    var entryUrl: String,
    @field:Json(name = "eid")
    var entryId: String
)
